cpp-package/example/test_ndarray.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ */
+#include <vector>
+#include "mxnet/c_api.h"
+#include "dmlc/logging.h"
+#include "mxnet-cpp/MxNetCpp.h"
+using namespace mxnet::cpp;
+
+enum TypeFlag {
+  kFloat32 = 0,
+  kFloat16 = 1,
+  kInt8 = 2

```

Comment:
Why the definition is different from the type flags in mshadow/base.h? Since it will be passed to the constructor of NDArray, I'm afraid you won't get what you want.
```cpp
enum TypeFlag {
  kFloat32 = 0,
  kFloat64 = 1,
  kFloat16 = 2,
  kUint8 = 3,
  kInt32 = 4,
  kInt8  = 5,
  kInt64 = 6,
};
```

Reply:
Originally, I just wanted to test if a new NDArray created by copy will have a type inconsistent issue and ignored the actual meaning of these type flags. Thanks for your reminding and will align with the definition in mshadow/base.h.